File: frontend/src/app/core/models/receita.model.ts
export interface ReceitaResponse {
  id: string;
  nome: string;
  valor: number;
  descricao?: string;
  arquivoId?: string;
  dataRecebimento: string;
  competencia: string;
  receitaTemplateId?: string;
  grupoId?: string;
  usuarioNome?: string;
  createdAt: string;
}

export interface ReceitaTemplateResponse {
  id: string;
  nome: string;
  valor: number;
  descricao?: string;
  ativa: boolean;
  dataInicio: string;
  dataFim?: string;
  createdAt: string;
}

export interface CriarReceitaRequest {
  usuarioId: string;
  nome: string;
  valor: number;
  descricao?: string;
  arquivoId?: string;
  dataRecebimento: string;
  receitaTemplateId?: string;
  grupoId?: string;
}

export interface CriarReceitaTemplateRequest {
  usuarioId: string;
  nome: string;
  valor: number;
  descricao?: string;
  dataInicio: string;
  dataFim?: string;
}

export interface AtualizarReceitaTemplateRequest {
  nome: string;
  valor: number;
  descricao?: string;
  dataInicio: string;
  dataFim?: string;
}

export interface GerarReceitasMesRequest {
  usuarioId: string;
  competencia: string;
}

File: backend/Ledger.Infrastructure/Data/Models/ReceitaModel.cs
namespace Ledger.Infrastructure.Data.Models;

public class ReceitaModel
{
    public Guid Id { get; set; }
    public string Nome { get; set; } = string.Empty;
    public decimal Valor { get; set; }
    public string? Descricao { get; set; }
    public Guid? ArquivoId { get; set; }
    public DateTime DataRecebimento { get; set; }
    public DateTime Competencia { get; set; }
    public Guid? ReceitaTemplateId { get; set; }
    public Guid UsuarioId { get; set; }
    public Guid? GrupoId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public virtual ArquivoModel? Arquivo { get; set; }
    public virtual ApplicationUser Usuario { get; set; } = null!;
    public virtual ReceitaTemplateModel? Template { get; set; }
    public virtual GrupoModel? Grupo { get; set; }
}
